lib/fluent/plugin/in_cloudwatch_logs.rb

```diff
@@ -84,18 +87,18 @@ def configure_parser(conf)
       end
     end
 
-    def state_file_for(log_stream_name)
-      return "#{@state_file}_#{log_stream_name.gsub(File::SEPARATOR, '-')}" if log_stream_name
-      return @state_file
+    def state_file_for(log_group_name, log_stream_name)
+      return "#{@state_file}_#{log_group_name.gsub(File::SEPARATOR, '-')}_#{log_stream_name.gsub(File::SEPARATOR, '-')}"
```

Comment:
What happened when old state file exists?
Is this breaking change?

If so, we should add migrating code for existing users.

Reply:
We could make it non breaking by just adding the log_group if the user specifies use_log_group_prefix

But whenever there's a switch state would be lost. Let me think about how to solve it.